src/Sql/Ddl/Column/AbstractPrecisionColumn.php

```diff
@@ -1,86 +1,60 @@
 <?php
 /**
- * Zend Framework (http://framework.zend.com/)
- *
- * @link      http://github.com/zendframework/zf2 for the canonical source repository
- * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
- * @license   http://framework.zend.com/license/new-bsd New BSD License
+ * @see       https://github.com/zendframework/zend-db for the canonical source repository
+ * @copyright Copyright (c) 2005-2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-db/blob/master/LICENSE.md New BSD License
  */
 
+declare(strict_types=1);
+
 namespace Zend\Db\Sql\Ddl\Column;
 
 abstract class AbstractPrecisionColumn extends AbstractLengthColumn
 {
-    /**
-     * @var int|null
-     */
+    /** @var int|null */
     protected $decimal;
 
-    /**
-     * {@inheritDoc}
-     *
-     * @param int|null $decimal
-     * @param int      $digits
-     */
     public function __construct(
-        $name,
-        $digits = null,
-        $decimal = null,
-        $nullable = false,
-        $default = null,
+        string $name = '',
+        ?int $digits = null,
+        ?int $decimal = null,
+        bool $nullable = false,
+        ?$default = null,
```

Comment:
hm... what is the question mark before variable name? what is the type?

Reply:
There are quite a few cases, where multiple argument types are defined in the DocBlock and not supported by PHP. I think this should be worked out to see if not a specific type can be agreed on in another task. In those cases I tried to keep the DocBlock with the proper definitions.

As defined in the [zcs 2.0](https://github.com/zendframework/zend-coding-standard/blob/2.0.0-alpha.3/docs/book/v2/coding-style-guide.md#classes-properties-and-methods)
>Nullable and optional arguments, which are marked as = null, must have the ? symbol present.

I interpret this as a must of the ? symbol, even if the type cannot be defined properly. Again, the goal would be to find one (1) specific common type and declare it as argument type.